pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -171,74 +202,279 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 		namespace string
 	}
 	{{ lower_camel .Name }}Chan := make(chan {{ lower_camel .Name }}ListWithNamespace)
+	var initial{{ upper_camel .Name }}List {{ .ImportPrefix }}{{ .Name }}List
+{{- end }}
+{{- end }}
 
-	var initial{{ upper_camel .Name }}List {{ .ImportPrefix }}{{ .Name }}List{{- end }}
+	currentSnapshot := {{ .GoName }}Snapshot{}
 
-{{- end}}
+{{- range .Resources}}
+{{- if not .ClusterScoped }}
+	{{ lower_camel .PluralName }}ByNamespace := sync.Map{}
+{{- end }}
+{{- end }}
 
-	currentSnapshot := {{ .GoName }}Snapshot{}
+	if ! watchNamespacesIsEmpty || opts.ExpressionSelector == "" {
```

Comment:
Why are we catching this if we are just ranging over watchNamespaces?
If watchNamespaces is empty then this does nothing as the range has no contents.

Reply:
`watchNamespaces` should be `len(watchedNamespaces) >= 1`

Reply:
gotcha